cli/waiter/subcommands/ssh.py

```diff
@@ -0,0 +1,199 @@
+    guard_no_cluster(clusters)
+    token_or_service_id_or_instance_id = args.pop('token-or-service-id-or-instance-id')
+    command = args.pop('command')
+    ssh_destination = args.pop('ssh_destination')
+    include_active_instances = args.pop('include_active_instances')
+    include_failed_instances = args.pop('include_failed_instances')
+    include_killed_instances = args.pop('include_killed_instances')
+    container_name = args.pop('container_name', 'waiter-app')
+    skip_prompts = args.pop('quick')
+    if ssh_destination == Destination.TOKEN:
+        return ssh_token(clusters, enforce_cluster, token_or_service_id_or_instance_id, command, container_name,
+                         skip_prompts, include_active_instances, include_failed_instances, include_killed_instances)
+    elif ssh_destination == Destination.SERVICE_ID:
+        return ssh_service_id(clusters, token_or_service_id_or_instance_id, command, container_name, skip_prompts,
+                              include_active_instances, include_failed_instances, include_killed_instances)
+    elif ssh_destination == Destination.INSTANCE_ID:
+        return ssh_instance_id(clusters, token_or_service_id_or_instance_id, command, container_name)
+
+
+def register(add_parser):
+    """Adds this sub-command's parser and returns the action function"""
+    parser = add_parser('ssh',
+                        help='ssh to a Waiter instance',
+                        description='ssh to an instance given the token, service id, or instance id. Working directory '
+                                    'will be the log directory.')
+    parser.add_argument('token-or-service-id-or-instance-id')
+    if is_admin_enabled():
+        parser.add_argument('--container-name', '-c',
+                            help='specify the container name you want to ssh into. Defaults to "waiter-app". Has no '
+                                 'effect if instance is not k8s pod.')
+    id_group = parser.add_mutually_exclusive_group(required=False)
+    id_group.add_argument('--token', '-t', dest='ssh_destination', action='store_const', const=Destination.TOKEN,
+                          default=Destination.TOKEN, help='Default; ssh with token')
+    id_group.add_argument('--service-id', '-s', dest='ssh_destination', action='store_const',
+                          const=Destination.SERVICE_ID, help='ssh using a service id')
+    id_group.add_argument('--instance-id', '-i', dest='ssh_destination', action='store_const',
+                          const=Destination.INSTANCE_ID, help='ssh directly to instance id')
+    parser.add_argument('--quick', '-q', dest='quick', action='store_true',
+                        help='skips services prompt by selecting the service with latest request, and instances prompt '
+                             'by selecting a random one.')
+    parser.add_argument('--active', '-a',  dest='include_active_instances', action='store_true', default=True,
```

Comment:
`active` and `no-active` belong in their own mutually exclusive group.
Same for failed/no-failed and killed/no-killed.